File: ketch-android/webview/Android Ketch Pref Center using Java/app/src/main/java/com/ketch/sample/webview/Consent.java
package com.ketch.sample.webview;

import androidx.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import java.util.List;
import java.util.Map;

public class Consent {
    @SerializedName("purposes")
    @Nullable
    public Map<String, Boolean> purposes;

    @SerializedName("vendors")
    @Nullable
    public List<String> vendors;

    @Override
    public String toString() {
        return "Consent{" +
                "purposes=" + purposes +
                ", vendors=" + vendors +
                '}';
    }
}

File: ketch-android/Android Ketch Pref Center using Kotlin/app/src/main/java/com/ketch/sample/prefcenter/Consent.kt
package com.ketch.sample.prefcenter

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Consent(
    @SerializedName("IABUSPrivacy_String")
    val IABUSPrivacy_String: String?,

    @SerializedName("IABTCF_TCString")
    val IABTCF_TCString: String?
): Parcelable
